### ISSUE TITLE
Previous Nodes are Not Working as Expected After Update.

### ISSUE THREAD

**Issue by happicamper** [chef/chef#8888]
## Description
I upgraded my Chef server and Chef client.
On Chef client 14, I can run this code freely on my macOS Mojave:
`sudo chef-client`
But now, after the confirmation about the license, it will break produce the output below.
`NoMethodError: undefined method `[]' for nil:NilClass`

It seems that node`['etc']['passwd'`] is no longer readable or has a value of nil when chef-client is executed.

## Chef Version
Chef 15.3.14

## Platform Version
macOS Mojave 10.14.4

## Replication Case
My previous nodes which are bootstrapped using chef-client 14 are now upgraded to 15. 
After executing `sudo chef-client` the existing recipe which works fine before is now broken.

## Client Output
```
-------------
undefined method `[]' for nil:NilClass

Cookbook Trace:
---------------
  /var/chef/cache/cookbooks/macos/recipes/mac_init.rb:62:in `from_file'

Relevant File Content:
----------------------
/var/chef/cache/cookbooks/macos/recipes/mac_init.rb:

 55:    action :create
 56:    owner 'administrator'
 57:    group 'localaccounts'
 58:    mode  '0775'
 59:  end
 60:
 61:  # Template for bash history
 62>>   node['etc']['passwd'].each do |user, data|
 63:
 64:    template "/Users/#{user}/.bash_profile" do
 65:      source '/default/bash_profile.erb'
 66:      owner 'administrator'
 67:      group 'staff'
 68:      ignore_failure true
 69:    end
 70:  end

```
```

```

**Comment by tas50** [comment]
Hey @happicamper we fixed a regression in Chef 15.3 where the Ohai Passwd plugin wasn't properly disabled. The Passwd plugin has historically caused a lot of issues for customers in large LDAP or AD setups. This means the data you're attempting to consume won't be available out of the box in Chef anymore. You'll need to enable the Passwd plugin in your Chef Infra Client  client.rb config now. There's some information about how to do that here: https://docs.chef.io/ohai.html#optional-plugins

**Comment by happicamper** [comment]
Thank you very much for your answer @tas50. It really shed some light on my issue.
